torch/csrc/jit/passes/mkldnn_rewrite.cpp

```diff
@@ -125,48 +171,153 @@ void insertMkldnnPrePackedOps(script::Module& module) {
-          %dilation:int[], %groups:int, %input_size:int[], %dummy_attr:str):
+    graph(${graph_input}
+          %input_size:int[], %attr_placeholder:str, %scalars_placeholder: Scalar?[], %algorithm_placeholder: str?${op_input_str}):
         %attr: str = prim::Constant[value="${op_attr}"]()
-        %packed_weight_bias : __torch__.torch.classes.mkldnn.ConvOpContext = mkldnn_prepacked::conv2d_prepack(
-            %weight, %bias, %stride, %padding, %dilation, %groups,
-            %input_size, %attr)
-        %res = mkldnn_prepacked::conv2d_run(%input, %packed_weight_bias)
+        ${construct_operand_list}
+        %packed_weight_bias : __torch__.torch.classes.${op_context_name} =  ${prepack_op_name}(
+            ${prepack_input}
+            %input_size, %attr, %scalars, %algorithm)
+        %res = ${run_op_name}(%input, %packed_weight_bias)
         return (%res))");
 
-  for (auto const& it : mkldnn::fusion_rewrite_map) {
+  for (auto const& it : fusion_rewrite_map) {
     std::string op = it.first;
-    if (op == std::string("none")) {
-      continue;
+    std::vector<std::string> op_input = it.second.scalar_input;
+    std::string algorithm_input = it.second.algorithm_input;
+
+    if (!algorithm_input.empty()) {
+      op_input.push_back(algorithm_input);
+    }
+    std::string op_input_str = c10::Join(", ", op_input);
+
+    if (!op_input.empty()) {
+      op_input_str = ", " + op_input_str;
     }
 
     at::jit::TemplateEnv env;
     env.s("op", op);
+    env.s("op_input_str", op_input_str);
+    env.s("prepack_op_name", prepack_op_name);
+    env.s("run_op_name", run_op_name);
+    env.s("graph_input", graph_input);
+    env.s("prepack_input", prepack_input);
 
     at::jit::TemplateEnv env_fused;
     env_fused.s("op_attr", op);
+    env_fused.s("op_input_str", op_input_str);
+    env_fused.s(
+        "construct_operand_list",
+        mkldnn::construct_operand_list(
+            it.second.scalar_input, it.second.algorithm_input));
+    env_fused.s("prepack_op_name", prepack_op_name);
+    env_fused.s("run_op_name", run_op_name);
+    env_fused.s("op_context_name", op_context_name);
+    env_fused.s("graph_input", graph_input);
+    env_fused.s("prepack_input", prepack_input);
 
     SubgraphRewriter rewriter;
     rewriter.RegisterRewritePattern(
         conv_op_rstring.format(env), conv_op_fused_rstring.format(env_fused));
 
-    auto filters = it.second;
+    auto filters = it.second.filters;
     rewriter.runOnGraph(graph, filters);
   }
 }
 
+void FuseEltwiseWithPackedOps(std::shared_ptr<Graph>& graph) {
+  RewriteEltwiseGraph(
+      graph,
+      mkldnn::fusion_rewrite_map(),
+      std::string("mkldnn_prepacked::conv2d_prepack"),
+      std::string("mkldnn_prepacked::conv2d_run"),
+      std::string("mkldnn.ConvOpContext"),
+      std::string(
+          "%input, %weight, %bias, %stride:int[], %padding:int[], %dilation:int[], %groups:int,"),
+      std::string("%weight, %bias, %stride, %padding, %dilation, %groups,"));
```

Comment:
These arguments are impossible to reason about. I understand that it's simply a substring later stitched into the pattern, but I would still recommend trying to make the parameters meaningful to avoid leaking abstractions.

Do you have another example of how this function will be used? If it's just one case, then we probably do not need it at all.